### PR TITLE
CORS-2590: Move to WWT environment for azurestack

### PR DESCRIPTION
**Pull request by rna-afk**
Reusing the WWT steps used by QE team to install a cluster on azurestack for the e2e-azurestack tests in order to move from the deprecated PPE environment to WWT.

### CODE REVIEW

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack pull-ci-openshift-installer-master-e2e-azurestack-upi

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack pull-ci-openshift-installer-master-e2e-azurestack-upi

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack pull-ci-openshift-installer-master-e2e-azurestack-upi

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack pull-ci-openshift-installer-master-e2e-azurestack-upi

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi pull-ci-openshift-installer-master-e2e-azurestack

**Comment by r4f4** [comment]
We need https://github.com/openshift/release/pull/40431 to fix the `e2e-azurestack-upi` failure:
```
 Installing python modules: yaml
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: module 'yaml' has no attribute 'full_load' 
```

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi pull-ci-openshift-installer-master-e2e-azurestack

**Comment by r4f4** [comment]
The cluster is installing, so in that sense it LGTM. However, the number of e2e failures seems very high:
* e2e-azurestack: `error: 3230 fail, 0 pass, 4 skip (2m51s)`
* e2e-azurestack-upi: `error: 3230 fail, 0 pass, 4 skip (2m37s)`
It seems all tests are panic'ing with
```
panic: interface conversion: cloudprovider.Interface is nil, not *azure.Cloud
```

**Comment by jhixson74** [comment]
> The cluster is installing, so in that sense it LGTM. However, the number of e2e failures seems very high:
> 
>     * e2e-azurestack: `error: 3230 fail, 0 pass, 4 skip (2m51s)`
> 
>     * e2e-azurestack-upi: `error: 3230 fail, 0 pass, 4 skip (2m37s)`
>       It seems all tests are panic'ing with
> 
> 
> ```
> panic: interface conversion: cloudprovider.Interface is nil, not *azure.Cloud
> ```

Any updates on this?

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi pull-ci-openshift-installer-master-e2e-azurestack

**Comment by rna-afk** [comment]
The error seems to be coming from another repo. Need to investigate there before we merge this.
https://github.com/openshift/origin/blob/e0aea0ed6b864859ad6efc6a7950f161ffd24d00/cmd/openshift-tests/provider.go#L67

**Comment by r4f4** [comment]
> The error seems to be coming from another repo. Need to investigate there before we merge this. https://github.com/openshift/origin/blob/e0aea0ed6b864859ad6efc6a7950f161ffd24d00/cmd/openshift-tests/provider.go#L67

It seems something is going wrong when loading the Cloud details here https://github.com/openshift/origin/blob/e0aea0ed6b864859ad6efc6a7950f161ffd24d00/vendor/k8s.io/kubernetes/test/e2e/framework/providers/azure/azure.go#L48-L50

I wonder if the endpoint `'https://management.mtcazs.wwtatc.com'` is returning something the k8s config loading is not expecting, thus resulting in an empty/invalid cloud.

**Comment by r4f4** [comment]
@rna-afk if you can deploy clusters in WWT, I suggest starting up a cluster and then running the `openshift/origin` tests under a debugger so you can inspect what's going on during the test initialization.

**Comment by r4f4** [comment]
You can also change this part https://github.com/openshift/origin/blob/e0aea0ed6b864859ad6efc6a7950f161ffd24d00/vendor/k8s.io/kubernetes/test/e2e/framework/providers/azure/azure.go#L48 to actually check the `err` from `azure.NewCloud` before trying to cast the return to `azure.Cloud`.

**Comment by rna-afk** [comment]
>I wonder if the endpoint 'https://management.mtcazs.wwtatc.com' is returning something the k8s config loading is not expecting, thus resulting in an empty/invalid cloud.

Yeah that's what I figured would be the problem. I felt the switch case [here](https://github.com/openshift/origin/blob/e0aea0ed6b864859ad6efc6a7950f161ffd24d00/cmd/openshift-tests/provider.go#L94) maybe needs 'azurestack'

**Comment by r4f4** [comment]
> > I wonder if the endpoint 'https://management.mtcazs.wwtatc.com' is returning something the k8s config loading is not expecting, thus resulting in an empty/invalid cloud.
> 
> Yeah that's what I figured would be the problem. I felt the switch case [here](https://github.com/openshift/origin/blob/e0aea0ed6b864859ad6efc6a7950f161ffd24d00/cmd/openshift-tests/provider.go#L94) maybe needs 'azurestack'

From the logs, it's being passed `none` so that wouldn't help: 
```
+ openshift-tests run openshift/conformance/parallel --provider none -o /logs/artifacts/e2e.log --junit-dir /logs/artifacts/junit
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38670/rehearse-38670-pull-ci-openshift-installer-master-e2e-azurestack/1677331043788525568/artifacts/e2e-azurestack/openshift-e2e-test/build-log.txt

**Comment by r4f4** [comment]
Here is where it's being set: https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh#L140-L143

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi pull-ci-openshift-installer-master-e2e-azurestack

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi pull-ci-openshift-installer-master-e2e-azurestack

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack

**Comment by r4f4** [comment]
@rna-afk I think we can drop the WIP on this.

File: ci-operator/step-registry/ipi/conf/azurestack/ipi-conf-azurestack-ref.yaml [hunk 1]
```diff
@@ -25,12 +25,17 @@ ref:
...
-    default: 'https://management.ppe3.stackpoc.com'
+    default: 'https://management.mtcazs.wwtatc.com'
     documentation: |-
       The Azure Stack Hub Endpoint to connect to.
   - name: SUFFIX_ENDPOINT
-    default: "ppe3.stackpoc.com"
+    default: "mtcazs.wwtatc.com"
     documentation: |-
       The endpoint suffix for storage accounts.
+  - name: CLUSTER_OS_IMAGE
```
Comment (jinyunma) [inline ci-operator/step-registry/ipi/conf/azurestack/ipi-conf-azurestack-ref.yaml:35]: this parameter seems not used in step `ipi-conf-azurestack`, I guess we don't need to define it here.

File: ci-operator/step-registry/ipi/conf/azurestack/rhcos/ipi-conf-azurestack-rhcos-ref.yaml [hunk 2]
```diff
@@ -12,11 +12,11 @@ ref:
     mount_path: /var/run/azurestack-cluster-secrets
   env:
   - name: RHCOS_VHD_RESOURCE_GROUP
-    default: "rhcos-storage-rg"
+    default: "rhcosvhd"
     documentation: |-
       The resource group of boot image vhd, the default value is for PPE3 environment.
```
Comment (jinyunma) [inline ci-operator/step-registry/ipi/conf/azurestack/rhcos/ipi-conf-azurestack-rhcos-ref.yaml:17]: How about to update PPE3 -> WWT in documentation? same comments for documentation of `RHCOS_VHD_STORAGE_ACCOUNT`

File: ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh [hunk 3]
```diff
@@ -137,9 +137,10 @@ aws|aws-arm64)
     export KUBE_SSH_USER=core
     ;;
 azure4|azure-arm64) export TEST_PROVIDER=azure;;
-azurestack)
+azurestack|azurestack-qe)
```
Comment (jinyunma) [inline ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh:140]: The cluster type for cluster profile `azurestack-qe` is `azurestack`, see code https://github.com/openshift/ci-tools/blob/master/pkg/api/types.go#L1372-L1375, so I think there is no need to add `azurestack-qe` here.

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi pull-ci-openshift-installer-master-e2e-azurestack

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by jinyunma** [comment]
@rna-afk I have another concern which we discussed before on slack, are those CI jobs running under QE account temporarily or permanently? 
 And not sure if we share the same subscription on WWT? If it is shared, we might need to adjust slice quota for cluster-profile `azurestack-qe`. Currently, only two slice quota are available under this cluster profile (https://github.com/openshift/release/blob/master/core-services/prow/02_config/_boskos.yaml#L939-L943)

cc @jianlinliu @gpei to be noticed and comments if any.

**Comment by rna-afk** [comment]
@jinyunma @jianlinliu Would it be alright if we could make this sit in the QE environment permanently? I can change it to azurestack-wwt and add leases.

**Comment by jianlinliu** [comment]
> Would it be alright if we could make this sit in the QE environment permanently

Before that, the first thing is we need to confirm QE and Dev is using the same project.

**Comment by rna-afk** [comment]
@patrickdillon do you know if QE and dev work on the same project in WWT?

**Comment by patrickdillon** [comment]
> @patrickdillon do you know if QE and dev work on the same project in WWT?

I'm not certain. I believe there are two subscriptions in WWT, but I'm not sure how they are shared. We can find out.

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 4]
```diff
@@ -4383,7 +4385,6 @@ presubmits:
       pj-rehearse.openshift.io/can-be-rehearsed: "true"
     name: pull-ci-openshift-installer-master-images
     rerun_command: /test images
-    skip_if_only_changed: (^docs/)|((^|/)OWNERS(_ALIASES)?$)|((^|/)[A-Z]+\.md$)
```
Comment (sadasu) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml:4386]: Why are we removing this? It is related to the change to `always_run` but I don't understand the motivation behind this change and its relation to the move to WWT environment for azurestack.

**Comment by sadasu** [comment]
/retitle: CORS-2590: Move to WWT environment for azurestack

**Comment by sadasu** [comment]
/retitle CORS-2590: Move to WWT environment for azurestack

**Comment by sadasu** [comment]
/retitle [CORS-2590](https://issues.redhat.com//browse/CORS-2590): Move to WWT environment for azurestack

**Comment by sadasu** [comment]
/retitle CORS-2590: Move to WWT environment for azurestack

**Comment by patrickdillon** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack

**Review by r4f4 — COMMENTED** [review]
There seem to be some unrelated changes mixed in with the move to WWT?

**Comment by rna-afk** [comment]
>There seem to be some unrelated changes mixed in with the move to WWT?

Fixed it thanks! Must've creeped in when I ran the make jobs a long time ago

**Comment by r4f4** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack

**Review by r4f4 — COMMENTED** [review]
Cluster installed.
/lgtm

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi

**Comment by rna-afk** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack
/pj-rehearse pull-ci-openshift-installer-master-e2e-azurestack-upi

**Comment by r4f4** [comment]
```
INFO[2023-10-24T19:03:42Z] Running step e2e-azurestack-ipi-deprovision-deprovision. 
INFO[2023-10-24T19:15:50Z] Step e2e-azurestack-ipi-deprovision-deprovision succeeded after 12m7s. 
INFO[2023-10-24T19:15:50Z] Running step e2e-azurestack-ipi-deprovision-azure-resourcegroup. 
INFO[2023-10-24T19:17:12Z] Step e2e-azurestack-ipi-deprovision-azure-resourcegroup succeeded after 1m22s. 
```
/lgtm

**Comment by r4f4** [comment]
/assign @patrickdillon 
for the deprovision approval.

**Pull request merged**